src/value.h

```diff
@@ -18,18 +18,24 @@
  * differing types is used to maintain the Cypher-defined global sort order
  * in the SIValue_Order routine. */
 typedef enum {
-	T_NULL = 0,
-	T_STRING = 0x001,
-	// T_INT32 = 0x002, // unused
-	T_INT64 = 0x004,
-	// T_UINT = 0x008, // unused
-	T_BOOL = 0x010, // shares 'longval' representation in SIValue union
-	// T_FLOAT = 0x020, // unused
-	T_DOUBLE = 0x040,
-	T_PTR = 0x080,
-	T_CONSTSTRING = 0x100, // only used in deserialization routine
-	T_NODE = 0x200,
-	T_EDGE = 0x400,
+	T_MAP = (1 << 0),
+	T_NODE = (1 << 1),
+	T_EDGE = (1 << 2),
+	T_ARRAY = (1 << 3),
+	T_PATH = (1 << 4),
+	T_DATETIME = (1 << 5),
+	T_LOCALDATETIME = (1 << 6),
+	T_DATE = (1 << 7),
+	T_TIME = (1 << 8),
+	T_LOCALTIME = (1 << 9),
+	T_DURATION = (1 << 10),
+	T_STRING = (1 << 11),
+	T_BOOL = (1 << 12), // shares 'longval' representation in SIValue union
+	T_INT64 = (1 << 13),
+	T_DOUBLE = (1 << 14),
+	T_NULL = (1 << 15),
+	T_PTR = (1 << 16),
```

Comment:
maybe I have missed it, what is the context that T_PTR is being used, besides of SIType_ToString?

Reply:
You're right, it's not used at all right now! We could remove it, though I can also imagine a pointer type being useful here later. I have no strong preference.